File: backend/Services/Parsers/ColorfleHardModeParser.cs
using System.Text.RegularExpressions;

namespace backend.Services.Parsers
{
    public class ColorfleHardModeParser : ColorfleParser
    {
        private ILogger<ColorfleHardModeParser> _logger;

        public ColorfleHardModeParser(ILogger<ColorfleHardModeParser> logger) : base(logger)
        {
            _logger = logger;
        }

        public override string GameName => "Colorfle (Hard mode)";
        public override string? HelpText => "Select 'HARD' from the difficulty selector.";
        protected override Regex Parser => new Regex($@"Colorfle \d+ \(Hard mode\) (?<{ScoreGroup}>[\d|X])/\d");
    }
}

File: lambdas/src/parsers/colorfle-hardmode-parser.ts
import { BasicScoreResultParser } from './base/basic-score-parser';

export class ColorfleHardModeParser extends BasicScoreResultParser {
  readonly countWinner = true;
  readonly gameName = 'Colorfle (Hard mode)';
  readonly helpText = "Select 'HARD' from the difficulty selector.";
  protected readonly extraContent = null;
  readonly url = 'https://colorfle.com/';
  
  // Matches "Colorfle 767 (Hard mode) X/6"
  // Note: C# uses [\d|X] which incorrectly includes literal pipe - fixed here to [\dX]
  protected readonly parser = new RegExp(
    `Colorfle \\d+ \\(Hard mode\\) (?<${BasicScoreResultParser.SCORE_GROUP}>[\\dX])/\\d`
  );
}
